### PR TITLE
Feat/create create2

### PR DESCRIPTION
**Pull request by roynalnaruto**
### Description

[_PR description_]

### Issue Link

https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/1130

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Contents

- [_item_]

### Rationale

[_design decisions and extended information_]

### How Has This Been Tested?

[_explanation_]

<hr>

## How to fill a PR description 

Please give a concise description of your PR.

The target readers could be future developers, reviewers, and auditors. By reading your description, they should easily understand the changes proposed in this pull request.

MUST: Reference the issue to resolve

### Single responsability

Is RECOMMENDED to create single responsibility commits, but not mandatory.

Anyway, you MUST enumerate the changes in a unitary way, e.g.

```
This PR contains:
- Cleanup of xxxx, yyyy
- Changed xxxx to yyyy in order to bla bla
- Added xxxx function to ...
- Refactored ....
```

### Design choices

RECOMMENDED to:
- What types of design choices did you face?
- What decisions you have made?
- Any valuable information that could help reviewers to think critically

### CODE REVIEW

File: zkevm-circuits/src/evm_circuit/execution/create.rs [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+
+        let callee_is_success = cb.query_bool();
+
+        let create = ContractCreateGadget::construct(cb);
+
+        cb.stack_pop(value.expr());
+        cb.stack_pop(init_code.offset_rlc());
+        cb.stack_pop(init_code.length_rlc());
+        cb.condition(IS_CREATE2.expr(), |cb| {
+            cb.stack_pop(create.salt_rlc());
```
Comment (lispc) [inline zkevm-circuits/src/evm_circuit/execution/create.rs:100]: "salt_rlc" here is keccak rlc, not word rlc. Line100 is wrong.

File: zkevm-circuits/src/evm_circuit/execution/create.rs [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+
+        let value = cb.query_word_rlc();
+
+        let init_code_memory_offset = cb.query_cell_phase2();
+        let init_code_length = cb.query_word_rlc();
+        let init_code =
+            MemoryAddressGadget::construct(cb, init_code_memory_offset, init_code_length);
+
+        let keccak_output = cb.query_word_rlc();
+        let new_address_rlc = cb.word_rlc::<N_BYTES_MEMORY_ADDRESS>(
```
Comment (lispc) [inline zkevm-circuits/src/evm_circuit/execution/create.rs:80]: N_BYTES_MEMORY_ADDRESS is 5, should be 20 here?

**Comment by hero78119** [comment]
Hi @roynalnaruto , is this pr ready to review or still got some ongoing tasks?

**Comment by roynalnaruto** [comment]
@hero78119 I will create a new pull request with the updated changes. Those changes were recently merged in Scroll's fork and include the handling of insufficient caller balance.

**Comment by roynalnaruto** [comment]
Closing this in favour of #1355

**Pull request closed without merging**